keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +163,222 @@ Running this set of tests is optional for a per-provider basis. We would
+  test to add more feature coverage if needed.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**  
+  - The rollout can fail if the ordering of CSIMigration{cloud-provider} flag was wrongly enabled on kubelet and kube-controller-manager. Specifically, if on the node side kubelet enables the flag and control-plane side the flag is not enabled, then the volume will not be able to be mounted successfully. 
+    - For workloads that running on nodes have not enable CSI migration, those pods will not be impacted. 
+    - For any pod that is being rescheduled to nodes that turned on migration during node upgrading, the volume mount will fail and pod will not come up correctly.
+  - Additionally, CSI Migration has a strong dependency on CSI drivers. So if the in-tree corresponding CSI driver is not properly installed, any volume related operation could fail.
+  - If feature parity is not guaranteed or if any bug exists in the CSI driver/csi-translation-lib, the rollout could fail because pod using the PV could fail to execute provision/delete/attach/detach/mount/unmount/resize operations depend on the bug itself.
+
+* **What specific metrics should inform a rollback?**
+  We have metrics on the CSI sidecar side called `csi_operation_duration_seconds` and core k8s metrics on both kube-controller-manager and kubelet side called `storage_operation_duration_seconds`. 
+  Both of them will have a `migrated` field to indicate whether indicate this operation is a migrated PV operation. 
+    - For `csi_operation_duration_seconds`, we will have a `grpc_status` field
+    - For `storage_operation_duration_seconds`, we will have a `status` field
+  
+  If the error ratio of these two metrics has an unusual strike or is keeping at a relatively higher level compared to in-tree model, it means something went wrong and we need a rollback.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+To turn it on by default in Beta, we require each in-tree plugin to at least manually test the upgrade->downgrade->upgrade path.
+For GA, we require such test exists in each driver's test CI.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+There will not be API removal in CSI migration itself. But eventually when CSI migration is all finished. We will plan to remove all in-tree plugins.
+So we will have in-tree plugin deprecated when CSIMigration{cloud-provider} goes to beta. And code removal will be required eventually.
+In addition, some CSI drivers are not able to maintain 100% backwards compatibility, so those drivers need to deprecate certain behaviors. For example, azurefile and vSphere has some corner cases that will not work anymore. (TODO: add more details.)
```

Comment:
It would be really useful to describe them here (or link to a place that is describing those).

Also - are these azufilre and vSphere the only cases you're ware?

Reply:
For vsphere, you can link to https://github.com/kubernetes/kubernetes/pull/98546

@gnufied can you open an issue for to track the azurefile deprecation?

The remaining plugins I'm not aware of any incompatibilities.

Reply:
Added